File: examples/async/aster_ws_public.py
"""Read Aster public WebSocket trade events."""

import asyncio

from dcex.ws import aster


async def main() -> None:
    """Subscribe to Aster futures aggregate trades and print two messages."""
    async with aster.public(market="futures") as ws:
        await ws.subscribe_agg_trades("BTC-USDT-SWAP")
        print(await ws.recv())
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/aster_ws_public.rs
use std::time::Duration;

use dcex::exchanges::aster::AsterMarket;
use dcex::ws::aster::AsterPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = AsterPublicWebSocket::new(AsterMarket::Futures, Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_agg_trades("BTC-USDT-SWAP").await?;
    println!("{}", ws.recv().await?);
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
